src/common_types/src/datum.rs

```diff
@@ -765,6 +776,11 @@ impl Datum {
         }
     }
 
+    fn valid_is_utf8(s: &str) -> Result<()> {
```

Comment:
This check may be expensive for long string, better to add an option to decide whether to do this check.